common/docker/client.go

```diff
@@ -157,14 +157,35 @@ func (pending pendingStarts) finish(id string) {
 }
 
 func (c *Client) AllContainerIDs() ([]string, error) {
-	all, err := c.ListContainers(docker.ListContainersOptions{All: true})
+	return c.containerIDs()
+}
+
+// ExistingContainerIDs returns a list of container IDs which either are running or
+// might be started in the future.
+func (c *Client) ExistingContainerIDs() ([]string, error) {
+	return c.containerIDs("running", "paused", "exited", "created", "restarting")
```

Comment:
> Is it worth making this distinction?

Unfortunately we have to, because for `dead` containers there is no corresponding `ContainerDestroyed` event which handler resets the `stopped` flag.

> Having an inclusive list like this is fragile

Totally agree. An alternative approach would be to iterate and inspect each container during the restoration. However, it does not eliminate the dependency on a container state.